app/common/validation/PropertyValidation.scala

```diff
@@ -48,4 +48,20 @@ object PropertyValidation {
         broughtForwardLosses, annualExemptAmount, disposalDate)))
     }
   }
+
+  def validatePropertyTaxOwed(propertyCalculateTaxOwedModel: PropertyCalculateTaxOwedModel): Either[String, PropertyCalculateTaxOwedModel] = {
+    val propertyChargeableGainModel = validatePropertyChargeableGain(propertyCalculateTaxOwedModel.propertyChargeableGainModel)
+    val previousTaxableGain = validateOptionDouble(propertyCalculateTaxOwedModel.previousTaxableGain, residentPropertyKeys.previousTaxableGain)
+    val previousIncome = validateDouble(propertyCalculateTaxOwedModel.previousIncome, residentPropertyKeys.previousIncome)
+    val disposalDate = validateDisposalDate(propertyCalculateTaxOwedModel.propertyChargeableGainModel.disposalDate)
+    val personalAllowance = disposalDate match {
+      case Right(date) => validateResidentPersonalAllowance (propertyCalculateTaxOwedModel.personalAllowance, date)
+      case Left(_) => Right(propertyCalculateTaxOwedModel.personalAllowance)
```

Comment:
I know if the validation on the disposalDate has failed then it doesn't matter what is really returned here but returning a Right (which in our code is used for successes). Seems strange. Maybe it would be better to return a Left("disposalDate validation has failed")?